packages/webapp/pages/feeds/new.tsx

```diff
@@ -0,0 +1,210 @@
+import { webappUrl } from '@dailydotdev/shared/src/lib/constants';
+import { ActionType } from '@dailydotdev/shared/src/graphql/actions';
+import { mainFeedLayoutProps } from '../../components/layouts/MainFeedPage';
+import { getLayout } from '../../components/layouts/MainLayout';
+import { defaultOpenGraph, defaultSeo } from '../../next-seo';
+
+type NewFeedFormProps = {
+  name: string;
+};
+
+const newFeedId = 'new';
+
+const discardPrompt: PromptOptions = {
+  title: labels.feed.prompt.discard.title,
+  description: labels.feed.prompt.discard.description,
+  okButton: {
+    title: labels.feed.prompt.discard.okButton,
+    color: ButtonColor.Ketchup,
+  },
+};
+
+const NewFeedPage = (): ReactElement => {
+  const queryClient = useQueryClient();
+  const router = useRouter();
+  const { completeAction } = useActions();
+  const { createFeed } = useFeeds();
+  const { displayToast } = useToastNotification();
+  const { showPrompt } = usePrompt();
+  const { FeedPageLayoutComponent } = useFeedLayout();
+
+  const { user } = useAuthContext();
+  const { feedSettings } = useFeedSettings({ feedId: newFeedId });
+  const [isPreviewFeedVisible, setPreviewFeedVisible] = useState(false);
+  const isPreviewFeedEnabled = feedSettings?.includeTags?.length >= 1;
+
+  const {
+    finishedOnboarding,
+    onFinishedOnboarding,
+    postOnboardingRedirect,
+    isAnimating,
+  } = useOnboardingAnimation();
+
+  const onValidateAction = () => {
+    return !isPreviewFeedEnabled;
+  };
+
+  const { onAskConfirmation } = useExitConfirmation({
+    message: discardPrompt.description as string,
+    onValidateAction,
+  });
+
+  const {
+    mutateAsync: onSubmit,
+    data: newFeed,
+    isLoading,
+  } = useMutation(
+    async ({ name }: NewFeedFormProps) => {
+      const result = await createFeed({ name });
+
+      await request(graphqlUrl, ADD_FILTERS_TO_FEED_MUTATION, {
+        feedId: result.id,
+        filters: {
+          includeTags: feedSettings?.includeTags || [],
+        },
+      });
+
+      return result;
+    },
+    {
+      onSuccess: (data) => {
+        queryClient.removeQueries(getFeedSettingsQueryKey(user, newFeedId));
+
+        onAskConfirmation(false);
+        onFinishedOnboarding();
+        postOnboardingRedirect(`${webappUrl}feeds/${data.id}`);
+      },
+      onError: () => {
+        displayToast(labels.error.generic);
+      },
+    },
+  );
+
+  useEffect(() => {
+    completeAction(ActionType.CustomFeedBeta);
+  }, [completeAction]);
+
+  if (finishedOnboarding) {
+    return (
+      <PreparingYourFeed
+        text={`Preparing ${newFeed?.flags?.name || 'your custom feed'}...`}
+        isAnimating={isAnimating}
+      />
+    );
+  }
+
+  const seo: NextSeoProps = {
+    title: 'Create custom feed',
+    openGraph: { ...defaultOpenGraph },
+    ...defaultSeo,
+  };
+
+  return (
+    <>
+      <NextSeo {...seo} />
+      <FeedPageLayoutComponent className={classNames('!p-0')}>
```

Comment:
Isn't this page and edit almost the same? Would it make sense to extract to a component?
Though not sure myself I think it's definitely clearer this way

Reply:
I tried but it just became one big component with a bunch of props. So I opted into extracting in smaller parts and only leaving different stuff in each page.

reuse:
- useFeeds - contains feed CRUD logic
- FeedCustomPreivew - preview logic
- FeedCustompreviewControls - preview controls

each page (new and edit):
- custom onSubmit implementation 
- custom validate/dirty
- custom redirect flow
- custom copy

Hope it makes sense?

Reply:
Yeah yeah, had a doubt about these that why my message above also mentioned this:

> Though not sure myself I think it's definitely clearer this way